### PR TITLE
Add sleep time of 1s in lineage tests

### PR DESCRIPTION
**Pull request by stisac**
*Issue #, if available:* https://sim.amazon.com/issues/AML-97027

*Description of changes:* Remove  pytest test skips in association integration tests

*Testing done:* 

tox -e py39 -- tests/unit/sagemaker/lineage

collected 45 items

tests/unit/sagemaker/lineage/test_action.py ........... [ 24%]
tests/unit/sagemaker/lineage/test_artifact.py ............ [ 51%]
tests/unit/sagemaker/lineage/test_association.py ....... [ 66%]
tests/unit/sagemaker/lineage/test_context.py ........... [ 91%]
tests/unit/sagemaker/lineage/test_dataset_artifact.py . [ 93%]
tests/unit/sagemaker/lineage/test_endpoint_context.py . [ 95%]
tests/unit/sagemaker/lineage/test_model_artifact.py . [ 97%]
tests/unit/sagemaker/lineage/test_visualizer.py . [100%]

-> tox -e py39 -- tests/integ/sagemaker/lineage

collected 28 items

tests/integ/sagemaker/lineage/test_action.py ....... [ 25%]
tests/integ/sagemaker/lineage/test_artifact.py ........ [ 53%]
tests/integ/sagemaker/lineage/test_association.py sss [ 64%]
tests/integ/sagemaker/lineage/test_context.py ....... [ 89%]
tests/integ/sagemaker/lineage/test_dataset_artifact.py . [ 92%]
tests/integ/sagemaker/lineage/test_endpoint_context.py . [ 96%]
tests/integ/sagemaker/lineage/test_model_artifact.py . [100%]

### CODE REVIEW

**Comment by stisac** [comment]
The failure from 'sagemaker-python-sdk-pr' tests are resulting from tests/integ/test_multidatamodel.py and is unrelated to the changes made in this pull-request.

**Comment by ahsan-z-khan** [comment]
> The failure from 'sagemaker-python-sdk-pr' tests are resulting from tests/integ/test_multidatamodel.py and is unrelated to the changes made in this pull-request.

Looking into this. Seems like last couple PR is failing due to this test.

File: tests/integ/sagemaker/lineage/test_association.py [hunk 1]
```diff
@@ -54,8 +53,29 @@ def test_list(association_objs, sagemaker_session):
...
+    association_obj.set_tags(tags)
+
+    while True:
+        actual_tags = sagemaker_session.sagemaker_client.list_tags(
+            ResourceArn=association_obj.source_arn
+        )["Tags"]
+        if actual_tags:
+            break
+    assert len(actual_tags) == 1
+    assert actual_tags == tags
```
Comment (danabens) [inline tests/integ/sagemaker/lineage/test_association.py:81]: these while true blocks at the very minimum need a time.sleep, ideally though we would follow best practices using  retry with backoff

Comment (stisac) [reply tests/integ/sagemaker/lineage/test_association.py:81]: Okay. Does a sleep time of 1 second sound good?

Also when I noticed the other tests for actions, artifacts or contexts, None of them had sleep in while true blocks and as such followed similar pattern.

Comment (danabens) [reply tests/integ/sagemaker/lineage/test_association.py:81]: 1 sec is good for now. Those other instances of `while True:` without sleep need to be updated.

Comment (stisac) [reply tests/integ/sagemaker/lineage/test_association.py]: Added, Thank you

**Comment by stisac** [comment]
The failure from 'sagemaker-python-sdk-pr' tests are resulting from tests/integ/test_multidatamodel.py and is unrelated to the changes made in this pull-request.

**Review by danabens — APPROVED** [review]

**Review by danabens — DISMISSED** [review]

**Review by danabens — APPROVED** [review]

**Comment by stisac** [comment]
The code changes made here is included in https://github.com/aws/sagemaker-python-sdk/pull/2141

So closing this PR as it doesn't require merge.
